### PR TITLE
Add --set-labels to register command to set labels at registration time on SCC

### PR DESCRIPTION
**Pull request by felixsch**
Add parameter `--set-labels` which allows setting multiple labels at registration time in SUSEConnect.

card: https://trello.com/c/RQuVrJEI/3581-mspel-assign-label-to-a-system-during-connect-registration

**NOTE:** This merge request needs to check if its behind a proxy and print a message that setting labels is not supported. Miquel is providing a PR soon. We should piggyback on it!

**How to test this merge request:**

```
$ cd <branch>
$ docker run --rm --privileged -ti --network host -v $(pwd):/connect registry.suse.com/bci/golang:1.21-openssl

> cd /connect
> git config --global --add safe.directory /connect
> go mod vendor
> make build
# expect: build succeeds

> ./out/suseconnect -r <regcode> --set-labels label1,label2,label3
# expect: registration succeeds, no label related output

# goto: https://scc.suse.com/organizations and check the org of your used subscription
# expect: system should be visible and has assigned labels

> ./out/suseconnect -d
> ./out/suseconnect -r <regcode> --set-labels label1,label2,label3,label4,label5,label6,label7,label8,label9,label10,label11
# expect: To see error showing that not more than 10 labels can be set
```

**As always, if you need help or have question regarding the merge request, do not hesitate to ask!**:rocket:

thank you

### CODE REVIEW

File: cmd/suseconnect/connectUsage.txt [hunk 1]
```diff
@@ -15,6 +15,10 @@ Manage subscriptions at https://scc.suse.com
                              be registered.
                              Relates that product to the specified subscription,
                              and enables software repositories for that product.
+        --set-labels [LABELS]
+                             Set labels in SCC when the product is registered.
+                             Multiple labels can be specified providing a comma
+                             seperated
```
Comment (arharovets) [inline cmd/suseconnect/connectUsage.txt:21]:
```suggestion
                             To add multiple labels, separate them with commas.
```

**Comment by arharovets** [comment]
I'm following the instructions provided in the description, and currently facing the following error on `make build`:
```
localhost:/connect # make build
go clean
mkdir -p out
go build -v -mod=vendor -buildmode=pie -o out/ github.com/SUSE/connect-ng/cmd/suseconnect
go: inconsistent vendoring in /connect:
	github.com/google/uuid@v1.6.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/stretchr/testify@v1.9.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/davecgh/go-spew@v1.1.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/pmezard/go-difflib@v1.0.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/stretchr/objx@v0.5.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	gopkg.in/yaml.v3@v3.0.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

	To ignore the vendor directory, use -mod=readonly or -mod=mod.
	To sync the vendor directory, run:
		go mod vendor
make: *** [Makefile:53: build] Error 1
```

Anything I can do to work around this?

**Comment by felixsch** [comment]
@arharovets Ah I missed adding the `vendoring`. Run `go mod vendor` before `make build`

**Review by arharovets — APPROVED** [review]

File: internal/connect/labels.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+var (
+	localSetLabels = setLabels
+)
+
+type Label struct {
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+}
+
+func AssignAndCrateLabels(labels []string) error {
```
Comment (mssola) [inline internal/connect/labels.go:17]: Create*

**Review by mssola — APPROVED** [review]

**Comment by likhitha77** [comment]
We might need to explicitly specify how we deal with whitespaces in label name. I believe spaces should be allowed since we allow it in SCC

I noticed that if there's any empty spaces in the label name, it only takes the first word. It also ignores all the labels that come after it too. See:
Screenshot 2024-10-08 at 7 33 52 PM

Screenshot 2024-10-08 at 7 34 29 PM



It also throws this error sometimes:
Screenshot 2024-10-08 at 7 39 13 PM

**Comment by arharovets** [comment]
> We might need to explicitly specify how we deal with whitespaces in label name. I believe spaces should be allowed since we allow it in SCC
> 
> I noticed that if there's any empty spaces in the label name, it only takes the first word. It also ignores all the labels that come after it too. See: Screenshot 2024-10-08 at 7 33 52 PM
> 
> Screenshot 2024-10-08 at 7 34 29 PM
> 
> It also throws this error sometimes: Screenshot 2024-10-08 at 7 39 13 PM

I see these as inevitable evil that accompanies CLI tools - we provide the instructions and instructions don't mention a whitespace. If the user adds a whitespace, they're on their own - this is a console tool so text formatting rules from text processing software cannot be applied :slightly_smiling_face:

File: internal/connect/labels.go (hunk 2 above)
Comment (felixsch) [reply internal/connect/labels.go]: fixed

**Pull request merged**